src/ModelContextProtocol/Utils/Json/McpJsonUtilities.cs

```diff
@@ -123,6 +123,61 @@ internal static bool IsValidMcpToolSchema(JsonElement element)
     [JsonSerializable(typeof(UnsubscribeRequestParams))]
     [JsonSerializable(typeof(IReadOnlyDictionary<string, object>))]
 
+    // Primitive types for use in consuming AIFunctions
+    [JsonSerializable(typeof(string))]
+    [JsonSerializable(typeof(byte))]
+    [JsonSerializable(typeof(byte?))]
+    [JsonSerializable(typeof(sbyte))]
+    [JsonSerializable(typeof(sbyte?))]
+    [JsonSerializable(typeof(ushort))]
+    [JsonSerializable(typeof(ushort?))]
+    [JsonSerializable(typeof(short))]
+    [JsonSerializable(typeof(short?))]
+    [JsonSerializable(typeof(uint))]
+    [JsonSerializable(typeof(uint?))]
+    [JsonSerializable(typeof(int))]
+    [JsonSerializable(typeof(int?))]
+    [JsonSerializable(typeof(ulong))]
+    [JsonSerializable(typeof(ulong?))]
+    [JsonSerializable(typeof(long))]
+    [JsonSerializable(typeof(long?))]
+    [JsonSerializable(typeof(nuint))]
+    [JsonSerializable(typeof(nuint?))]
+    [JsonSerializable(typeof(nint))]
+    [JsonSerializable(typeof(nint?))]
+    [JsonSerializable(typeof(bool))]
+    [JsonSerializable(typeof(bool?))]
+    [JsonSerializable(typeof(char))]
+    [JsonSerializable(typeof(char?))]
+    [JsonSerializable(typeof(float))]
+    [JsonSerializable(typeof(float?))]
+    [JsonSerializable(typeof(double))]
+    [JsonSerializable(typeof(double?))]
+    [JsonSerializable(typeof(decimal))]
+    [JsonSerializable(typeof(decimal?))]
+    [JsonSerializable(typeof(Guid))]
+    [JsonSerializable(typeof(Guid?))]
+    [JsonSerializable(typeof(Uri))]
+    [JsonSerializable(typeof(Version))]
+    [JsonSerializable(typeof(TimeSpan))]
+    [JsonSerializable(typeof(TimeSpan?))]
+    [JsonSerializable(typeof(DateTime))]
+    [JsonSerializable(typeof(DateTime?))]
+    [JsonSerializable(typeof(DateTimeOffset))]
+    [JsonSerializable(typeof(DateTimeOffset?))]
+#if NET
+    [JsonSerializable(typeof(DateOnly))]
+    [JsonSerializable(typeof(DateOnly?))]
+    [JsonSerializable(typeof(TimeOnly))]
+    [JsonSerializable(typeof(TimeOnly?))]
+    [JsonSerializable(typeof(Half))]
+    [JsonSerializable(typeof(Half?))]
+    [JsonSerializable(typeof(Int128))]
+    [JsonSerializable(typeof(Int128?))]
+    [JsonSerializable(typeof(UInt128))]
+    [JsonSerializable(typeof(UInt128?))]
```

Comment:
Out of curiosity, how did you generate this list? Did you just loop through everything registered with TypeConverter by default? Or everything that implements ISpanParsable?

Reply:
Basically all primitives plus things supported by ASP.NET and that looked useful / relevant.